src/components/Zoom/Zoom.js

```diff
@@ -53,112 +35,101 @@ const propTypes = {
 };
 
 const defaultProps = {
-  zoomInTitle: 'Zoom in',
-  zoomOutTitle: 'Zoom out',
-  className: 'tm-zooms-bar',
-  zoomInClassName: 'tm-button tm-round-blue',
-  zoomOutClassName: 'tm-button tm-round-blue',
+  titles: {
+    zoomIn: 'Zoom in',
+    zoomOut: 'Zoom out',
+  },
   zoomInChildren: <FaPlus focusable={false} />,
   zoomOutChildren: <FaMinus focusable={false} />,
   zoomSlider: false,
 };
 
+const updateZoom = (map, zoomAction) => {
+  map.getView().cancelAnimations();
+  const zoom = map.getView().getZoom();
+
+  map.getView().animate({
+    zoom: zoom + zoomAction,
+  });
+};
+
 /**
  * This component creates a zoom wrapper.
  */
-class Zoom extends Component {
-  constructor(props) {
-    super(props);
-    this.ref = React.createRef();
-  }
-
-  componentDidMount() {
-    const { zoomSlider } = this.props;
-    if (zoomSlider) {
-      this.displayZoomSlider();
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    const { zoomSlider } = this.props;
-    if (prevProps.zoomSlider !== zoomSlider) {
-      if (zoomSlider) {
-        this.displayZoomSlider();
-      } else {
-        this.removeZoomSlider();
+function Zoom({
+  map,
+  titles,
+  zoomInChildren,
+  zoomOutChildren,
+  zoomSlider,
+  ...other
+}) {
+  const ref = useRef();
+  const zoomIn = useCallback(
+    evt => {
+      if (!evt.which || evt.which === 13) {
+        updateZoom(map, 1);
       }
+    },
+    [map],
+  );
+
+  const zoomOut = useCallback(
+    evt => {
+      if (!evt.which || evt.which === 13) {
+        updateZoom(map, -1);
+      }
+    },
+    [map],
+  );
+
+  useEffect(() => {
+    let control;
+    if (zoomSlider && ref.current) {
+      control = new ZoomSlider();
+      control.element.classList.remove('ol-control');
```

Comment:
Why do we need this? A comment would be nice? Also the next line is unclear to me.

Reply:
To void the apply of some default ol css I guess. @edellucien do you remember why?

Reply:
I remove this line. But leave the enxt one it's to avoid TAB navigation on this button.